### ISSUE TITLE
Trace/breakpoint trap (core dumped)

### ISSUE THREAD

**Issue by MyoungHaSong** [HolmesShuan/Location-aware-Upsampling-for-Semantic-Segmentation#2]
Thank you for sharing code.

I prepared datasets and ran example training code.

but I got this error Trace/breakpoint trap(core dumped), and training was stopped.

<img width="565" alt="error" src="https://user-images.githubusercontent.com/28686925/72485740-12111200-384c-11ea-888a-3c68c334f8e4.PNG">

what should I do ?

**Comment by HolmesShuan** [comment]
Thanks for your interest. I did not get this error during my training/validation process (it is strange that core dumped occurred after 5 epochs). We have tested our codes in the following envs: 
```
1. TiTAN Xp + PyTorch1.0.0 + gcc 5.4.0 + CUDA9.0
2. GTX 1080Ti + PyTorch1.0.0 + gcc 5.4.0 + CUDA9.0
3. Tesla K80 + PyTorch1.0.0 + gcc 5.5.0 + CUDA9.1
```

Sorry for any inconvenience. I will let this issue open until it can be solved.

**Comment by MyoungHaSong** [comment]
My env is TiTAN Xp + pytorch 1.0.1.post2, + gcc 5.4.0 + CUDA10.0 
So I changed my torch version to  1.0.0 and I ran same training code, but I got this error 

<img width="1125" alt="error2" src="https://user-images.githubusercontent.com/28686925/72489458-71285400-3857-11ea-8a66-a15cf3db601e.PNG">

**Comment by HolmesShuan** [comment]
Could you please detail the training script?
We still recommend CUDA9.0, since we have not tested our codes on CUDA10.0. Our codes base on PyTorch-Encoding (SyncBN was introduced by Encoding), which may encounter some problems with CUDA10.0.

**Comment by MyoungHaSong** [comment]
here is detail training script.

CUDA_VISIBLE_DEVICES=0,1 python train.py --dataset pcontext \
    --model encnet --dilated --lateral --aux --se-loss --offset-loss --batch-size 8\
    --backbone resnet101 --checkname encnet_resnet101_pcontext --no-val --offset-weight 0.3 --location-weight 0.1 --up-factor 4 --batch-size-per-gpu 2 --bottleneck-channel 64 --offset-branch-input-channel 512 --category 150 --base-size 520 --crop-size 480 --downsampled-input-size 60

**Comment by HolmesShuan** [comment]
Since the batchsize is 8 and only 2 gpus were used, you need to change `--batch-size-per-gpu 2` to `--batch-size-per-gpu 4`.

**Comment by MyoungHaSong** [comment]
I changed my script but I got same error 
runtimeerror: shape '[4,64,240,240]' is invalid for input of size 4

**Comment by HolmesShuan** [comment]
Please refer to my pervious [comment](https://github.com/HolmesShuan/Location-aware-Upsampling-for-Semantic-Segmentation/issues/2#issuecomment-574963938). SyncBN error was caused by PyTorch-Encoding.

Could you please rerun the new command in your previous env (`TiTAN Xp + pytorch 1.0.1.post2, + gcc 5.4.0 + CUDA10.0`).

**Comment by MyoungHaSong** [comment]
Thank you for the answer to you. Currently it's working fine without problems up to 11epoch.

How do I set --batch-size-per-gpu if I set batch to 4 on 3 GPUs?

**Comment by HolmesShuan** [comment]
Unfortunately, the answer is batchsize must be **K x GPU number**, which means `batch-size-per-gpu==K`. Note that batchsize=1 means using a single image for training. We did not support to distribute 4 images to 3 GPUs so far.

For 3 GPUs, `--batch-size-per-gpu` can be 3/6/9/12 etc. For batch=4, GPU number should be 1/2/4.
